code/modules/clothing/head/_head.dm

```diff
@@ -15,6 +16,50 @@
 		var/mob/living/carbon/human/H = loc
 		H.update_hair()
 
+///Special throw_impact for hats to frisbee hats at people to place them on their heads/attempt to de-hat them.
+/obj/item/clothing/head/throw_impact(atom/hit_atom, datum/thrownthing/thrownthing)
+	. = ..()
+	///if the thrown object's target zone isn't the head	
+	if(thrownthing.target_zone != BODY_ZONE_HEAD)
+		return
+	///ignore any hats with the tinfoil counter-measure enabled
+	if(clothing_flags & ANTI_TINFOIL_MANEUVER)
+		return
+	if(iscarbon(hit_atom))
+		var/mob/living/carbon/H = hit_atom
+		if(istype(H.head, /obj/item))

```

Comment:
Usually not needed. You can reasonably assume that the head slot is an item, if it exists. Also it looks like you just broke the functionality that allowed throwing hats onto people that already don't have any.